src/Data/List/NonEmpty.purs

```diff
@@ -62,3 +65,10 @@ length (NonEmptyList (x :| xs)) = 1 + L.length xs
 
 concatMap :: forall a b. (a -> NonEmptyList b) -> NonEmptyList a -> NonEmptyList b
 concatMap = flip bind
+
+sort :: forall a. Ord a => NonEmptyList a -> NonEmptyList a
+sort xs = sortBy compare xs
+
+sortBy :: forall a. (a -> a -> Ordering) -> NonEmptyList a -> NonEmptyList a
+sortBy cmp xs = unsafeFromList $ L.sortBy cmp (toList xs)
```

Comment:
Another option here is to sort the tail and insert the head into the sorted result, a la insertion sort. That would require us to change the type of `insert` though, or to provide an alternative version.